index.js

```diff
@@ -15,32 +15,44 @@ var util = require('audio-buffer-utils');
  */
 function gain(options) {
 	// gain()
-	if (typeof options === 'undefined') options = 1;
+	if (typeof options === 'undefined') options = {volume: 1};
+
+	// gain(Function)
+	if (typeof options === 'function')  options = {volume: options};
 
 	// gain(Number)
 	// gain({volume: Number})
 	if (typeof options === 'number' || typeof options.volume === 'number') {
-		var volume = Math.tan(options.volume || options);
-		options = {volume: function (x) { return x * volume }};
-	}
+		var volume = options.volume || options;
 
-	// gain(Function)
-	else if (typeof options === 'function')  {
-		options = {volume: options};
+		// TODO: options.mode statements here
+		volume = Math.tan(volume);
+
+		options = {volume: volume};
 	}
 
 	// Return functions.
-	write.end = noop;
-	return write;
+	through.end = noop;
+	return through;
 
 	// Write volume to buffer
-	function write(buf) {
+	function through(buf) {
```

Comment:
Well, it would give unexpected errors if they did.  i.e. trying to call a number as a function, or coercing a function to a number or string in the math.